### ISSUE TITLE
The Known Issues of Az.Accounts 3.0.0 and Their Workarounds

### ISSUE THREAD

**Issue by msJinLei** [Azure/azure-powershell#25005]
## The interactive login method cannot popup window for WAM
Report an error **"User canceled authentication"**
- Workaround
```
Update-AzConfig -EnableLoginByWam $false
```
## Azure PowerShell cmdlets are unable to run the cmdlets after login with Username+Password or DeviceCode
-  Username+Password or DeviceCode login methods will fail without `-Tenant` parameter
- Workaround
1. Login interactively with the same account in the same machine 
2. If 1 is not applicable, disable WAM
```
Update-AzConfig -EnableLoginByWam $false
```

## WAM pop-up window to select an account is not easy to find
- Workaround: minimize other windows to look for the pop-up window carefully

**Comment by slavizh** [comment]
What about https://github.com/Azure/azure-powershell/issues/25003?

**Comment by msJinLei** [comment]
> What about #25003?

Thanks for reporting the issue. I reply in it.

**Comment by msJinLei** [comment]
> In ISE, on Windows 11 mine says the following, does this correlate?
> 
> ```powershell
> # 12.0.0    Az
> # 3.0.0     Az.Accounts
> # 6.0.0     Az.KeyVault
> Connect-AzAccount -Tenantid [TenantID] -Subscription [Subscription]
> # > Connect-AzAccount : InteractiveBrowserCredential authentication failed: A window handle must be configured. See https://aka.ms/msal-net-wam#parent-window-handles 
> # > Could not find tenant id for provided tenant domain '[TenantID]'. Please ensure that the provided user is found in the provided tenant domain.
> ```

Yes, we have reported your issue to MSAL https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4796

**Comment by spongemike2** [comment]
I get the same issue whenever I run from an elevated console window. Running the:

```
Update-AzConfig -EnableLoginByWam $false
```

...command fixes it.

**Comment by Alex-wdy** [comment]
[Az.Accounts 3.0.1](https://github.com/Azure/azure-powershell/releases/tag/v12.1.0-July2024)  We have solved the first issue in the latest version：
1. The interactive login method cannot popup window for WAM
We are still working on other issues and will update the progress gradually.

**Comment by RichardS013** [comment]
Using updated versions Az.Accounts 3.0.1 - 3.0.4 and it appears to still hang even after running cmdlet:
Update-AzConfig -EnableLoginByWam $false

**Comment by Alex-wdy** [comment]
> Using updated versions Az.Accounts 3.0.1 - 3.0.4 and it appears to still hang even after running cmdlet: Update-AzConfig -EnableLoginByWam $false

Can you give me some more details? Normally, this issue should not exist.

**Comment by Alex-wdy** [comment]
2 and 7 issues will be released at Az.13.0.0 version.

**Comment by ChristopherGLewis** [comment]
```
4. WAM pop-up window to select an account is not easy to find
   Workaround: minimize other windows to look for the pop-up window carefully
```

Is this for real? Your response to a fundamental break in the PowerShell login experience is a shrug?

Like most developers, I'm running a multi screen display environment, and *every* time I log in, it's like I'm playing wack-a-mole trying to find this *stupid* login screen.  Sometimes it's on the same screen, but buried behind the 8-10 windows I run.  Sometimes on my second or third monitor, but again NEVER on top.

How trivial would it be to add the "Always on top" (WS_EX_TOPMOST) flag to the window?  I believe that was implemented in Windows 3.x.

**Comment by msJinLei** [comment]
> it's like I'm playing wack-a-mole trying

I understand your pain-point. We list the "workaround" just to unblock the customers who thought the login window was not popped-up. We create the issue to track it. But the fix may refer to several teams so as to require more time.

**Comment by ChristopherGLewis** [comment]
> > it's like I'm playing wack-a-mole trying
> 
> I understand your pain-point. We list the "workaround" just to unblock the customers who thought the login window was not popped-up. We create the issue to track it. But the fix may refer to several teams so as to require more time.

It's been 5 months.  These teams are all Microsoft teams.

WAM should have *never* been released with this fundamental flaw.

**Comment by md7648** [comment]
WAM is a pile of shit - never works - always have to login via Device Code 

Disabling WAM also doesn't work, still get the WAM box popping up at this point.
